### PR TITLE
allow file system to customize all supported FUSE feature

### PR DESCRIPTION
**Pull request by bergwolf**

### CODE REVIEW

**Review by liubogithub — APPROVED** [review]

File: src/api/server/sync_io.rs [hunk 1]
```diff
@@ -625,39 +625,31 @@ impl<F: FileSystem + Sync, D: AsyncDrive> Server<F, D> {
...
-            | FsOptions::MAX_PAGES
-            | FsOptions::EXPLICIT_INVAL_DATA
-            | FsOptions::PERFILE_DAX;
-
         let capable = FsOptions::from_bits_truncate(flags);
 
         match self.fs.init(capable) {
             Ok(want) => {
-                let enabled = capable & (want | supported);
+                let enabled = capable & want;
```
Comment (xujihui1985) [inline src/api/server/sync_io.rs:632]: is there any impact for PassthroughFs? I saw it has implementation for Filesystem and if supported removed here the result enabled flags may be different

Comment (bergwolf) [reply src/api/server/sync_io.rs:632]: It still works as long as it is attached via vfs. It is broken if it is directly attached but that is not how we are using passthroughFS right now.

File: src/transport/mod.rs [hunk 2]
```diff
@@ -406,6 +407,13 @@ impl<S: BitmapSlice> io::Read for Reader<'_, S> {
     }
 }
 
+/// Safe wrapper for `sysconf(_SC_PAGESIZE)`.
+#[inline(always)]
+pub fn pagesize() -> usize {
+    // Trivially safe
+    unsafe { sysconf(_SC_PAGESIZE) as usize }
+}
```
Comment (jiangliu) [inline src/transport/mod.rs:415]: Will libc/musl cache the sysconf page size or will it cause a syscall?
If syscall is involved, should we use lazy_static() to store the value?

Comment (bergwolf) [reply src/transport/mod.rs:419]: fixed.

**Review by jiangliu — APPROVED** [review]

File: src/api/server/sync_io.rs (hunk 1 above)
Comment (xujihui1985) [reply src/api/server/sync_io.rs:632]: ok, may need some notice on release note for that change in next version

**Pull request merged**